src/iexamine.cpp

```diff
@@ -73,7 +73,7 @@ void iexamine::gaspump(player *p, map *m, const tripoint &examp)
                 item spill( item_it->type->id, calendar::turn );
                 const auto min = item_it->liquid_charges( 1 );
                 ///\EFFECT_DEX decreases amount of gas spilled from a pump
-                const auto max = item_it->liquid_charges( 1 ) * 8.0 / p->dex_cur;
+                const auto max = item_it->liquid_charges( 1 ) * 8.0 / std::max( 1, p->dex_cur );
```

Comment:
> What is your intended aim?

Well, I made the changes you suggested, then I was trying to 'update' the code a bit. This is what I have so far.
```C++
void iexamine::gaspump(player &p, const tripoint &examp)
{
  ...
    auto items = g->m.i_at( examp );
    for( item item_it : items ) {
        if( item_it.made_of(LIQUID) ) {
            if( one_in(10 + p.get_dex()) ) {
                add_msg(m_bad, _("You accidentally spill the %s."), item_it.type_name());
                item spill( item_it.typeId(), calendar::turn );
                const auto min = item_it.charges;
                const auto max = item_it.charges * 8.0 / std::max( 1, p.get_dex() );
                spill.charges = rng( min, max );
                g->m.add_item_or_charges( p.pos(), spill, 1 );
                item_it.charges -= spill.charges;
                if( item_it.charges < 1 ) {
                    items.erase(  item_it ); <==== I'm doing something wrong.
                }
            } else {
              ...
            return;
        }
    }
    add_msg(m_info, _("Out of order."));
}
```

I promise I won't take much more of you time (today at least :)